### PR TITLE
[#6999] Added unit test for irods::process_stash library (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
Passed in the testing environment.

### CODE REVIEW

File: unit_tests/src/test_process_stash.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#include <iterator>
+#include <string>
+
+TEST_CASE("process_stash basic operations")
+{
+    // A POD type used to demonstrate the stash's ability to store heterogeneous data.
+    struct pod
+    {
+        int x = 0;
+        int y = 0;
```
Comment (trel) [inline unit_tests/src/test_process_stash.cpp:17]: by heterogeneous... you mean... what?

there are both ints?    can you put different types here at the same time?

Comment (alanking) [reply unit_tests/src/test_process_stash.cpp:17]: I think it's referring to the stash, not the struct.

Comment (trel) [reply unit_tests/src/test_process_stash.cpp:17]: ohhh, found a parser :)

okay, so maybe a slight rephrasing of line 13 is in order...

Comment (korydraughn) [reply unit_tests/src/test_process_stash.cpp:17]: Will do. I felt that line would need a tweak.

Comment (korydraughn) [reply unit_tests/src/test_process_stash.cpp:17]: Updated.

**Comment by korydraughn** [comment]
Any remaining thoughts before I add the pound?

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
#'d

**Pull request merged**